src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+                return False
+        else:
+            return False
+
+    def _queue_latest_change(self):
+        """
+        This method is called when there's an input change.
+        If there's a change already being processed, it will override it and process the latest change.
+        """
+        with self.lock:  # Ensure thread-safe access to latest_change
+            self.latest_change = True  # Overwrite with the latest change
+            if self.is_processing:  # If a process is running, it will handle the latest change.
+                logging.warning("Multipoint Correlation is running. It will handle the latest change.")
+                self.correlation_txt.SetLabel("Correlation RMS Deviation :  Calculating...")
+                return
+            # Start processing the latest change in a separate thread
+            self.process_thread = threading.Thread(target=self._process_latest_change)
+            self.process_thread.start()
+
+    def _process_latest_change(self):
+        """
+        Processes only the latest change in targets.
+        """
+        while True:
+            with self.lock:
+                if not self.latest_change:
+                    break  # No more changes to process
+                self.latest_change = False  # Clear the latest change to indicate it's being processed
+                self.is_processing = True
+            # Process the change (simulate long processing)
+            self._do_correlation()
+            with self.lock:
+                rms = self.correlation_target.correlation_result["output"]["error"]["rms_error"]
+                self.correlation_txt.SetLabel(f"Correlation RMS Deviation : {rms}")
+                self.is_processing = False  # Mark that processing is complete
+
+    def _do_correlation(self):
+        """Run the correlation between the FIB and FM images."""
+        # Modify the input data to match the required format to run 3DCT
+        fm_das = [stream.raw[0] for stream in self.correlation_target.fm_streams]
+        fm_image = _convert_das_to_numpy_stack(fm_das)
+        fib_da = self.correlation_target.fib_stream.raw[0]
+        fib_coords = []
+        fm_coords = []
+        poi_coords = []
+        path = os.getcwd()
+        for fib_coord in self.correlation_target.fib_fiducials:
+            fib_coord = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(fib_coord.coordinates.value[0:2])
+            fib_coords.append(fib_coord)
+        fib_coords = numpy.array(fib_coords, dtype=numpy.float32)
+        for fm_coord in self.correlation_target.fm_fiducials:
+            fm_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(fm_coord.coordinates.value[0:2])
+            fm_coords.append([fm_coord_2d[0], fm_coord_2d[1], fm_coord.coordinates.value[2]])
+        fm_coords = numpy.array(fm_coords, dtype=numpy.float32)
+        min_n = min(len(fib_coords), len(fm_coords)) # make fm coords and fib coords the same length
```

Comment:
Rather than clipping, don't do correlation if they're not the same length. It's more likely to be an error (user forgot a point, added one by mistake, etc) than be useful